src/ClickOutsideHook.res

```diff
@@ -0,0 +1,41 @@
+@ocaml.doc("
+Check the element that is clicked or touched exist in the refs or not
+If it is yes do nothing
+Otherwise call function onClickOutside
+")
+let use = (~refs, ~onClick=true, ~onTouch=true, onClickOutside) => {
+  let listener = element =>
+    if (
+      !(
+        refs
+        ->Belt_Array.keepMap(({React.current: current}) => current->Js.Nullable.toOption)
+        ->Belt_Array.some(Webapi.Dom.Element.contains(element))
+      )
+    ) {
+      onClickOutside()
+    }
+
+  @ocaml.doc("
+  Takes a mouse event and return its target as an Html element
+  and apply listener when mouse click
+  ")
+  let mouseDownListener = event =>
+    event
+    ->Webapi.Dom.MouseEvent.target
+    ->Webapi.Dom.EventTarget.unsafeAsElement
+    ->{onClick ? listener : ignore}
+
+  @ocaml.doc("
+  Takes a touch event and return its target as an Html element
+  and apply listener when touch start
+  ")
+  let touchStartListener = event =>
```

Comment:
Since these functions are used once and only in this context maybe we can use a simple comment instead of ocaml.doc?